test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -170,6 +170,21 @@ Describe "Tests for parameter binding" -Tags "CI" {
         ( get-foo -b b a c d ) -join ',' | Should -BeExactly 'a,c,d'
     }
 
+    It 'Too many parameter sets defined' {
+        $scriptblock = {
+            param($numSets=1)
+            $parameters = (1..($numSets) | ForEach-Object { "[Parameter(parametersetname='set$_')]`$a$_" }) -join ', '
+            $body = "param($parameters) 'worked'"
+            $sb = [scriptblock]::Create($body)
+            & $sb -a1 123
+        }
+
+        { & $scriptblock -numSets 32 } | Should -Not -Throw
+        & $scriptblock -numSets 32 | Should -Be 'worked'
+        { & $scriptblock -numSets 33 } | Should -Throw -ErrorId 'ParsingTooManyParameterSets'
+        { & $scriptblock -numSets 34 } | Should -Throw -ErrorId 'ParsingTooManyParameterSets'
```

Comment:
I wonder why we would have the duplicates.
```suggestion
        & $scriptblock -numSets 32 | Should -Be 'worked'
        { & $scriptblock -numSets 33 } | Should -Throw -ErrorId 'ParsingTooManyParameterSets'

```

Reply:
32? It was to provide a better error message from `Should`, but I agree it's overkill.